### PR TITLE
Minimal working example

### PR DESCRIPTION
**Pull request by ghost**
It seems to be rather complicated to provide fine-tuning, but it also seems to be quite elastic.

### CODE REVIEW

**Comment by ghost** [comment]
I have a few doubts about how to set styles of subplots in appropriate way, btw.
They have to be really volatile. Maybe we should impose such sets (color, marker etc.) for different subplot numbers? Or should we generate them in some way?

File: concept.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import pandas as pd
+
+
+class Subplot:
+    def __init__(self, x, y):
+        self.x, self.y = x, y
+
+    def plot(self):
+        plt.plot(self.x, self.y)
+
```
Comment (gkorepanov) [inline concept.py:11]: Seems you have inserted the singleton idea into the concept. Looks nice.

File: concept.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    for subplot in plot.list:
+        subplot.plot()
+
+    if output:
+        plt.savefig(output, dpi=dpi, bbox_inches='tight')
+    if display:
+        plt.show()
+
+
+def add(input, x=0, y=1):
```
Comment (gkorepanov) [inline concept.py:26]: It's not clear how are you going to pass additional parameters (your idea you have told me in flesh was reasonable)
 I mean 
`add(mypandastable, x=0, y=1, xerr=2, yerr=3, {marker params}, {line params})`

File: concept.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+class Subplot:
+    def __init__(self, x, y):
+        self.x, self.y = x, y
+
+    def plot(self):
+        plt.plot(self.x, self.y)
+
+
+def plot(output=None, display=True, dpi=300):
```
Comment (gkorepanov) [inline concept.py:13]: I don't like such an api. It's quite non-intuitive.
The good variant would be probably having 
1. `plot(display=True, ...)`
2. `display(...)`
3. `save(...)`

**Review by gkorepanov — COMMENTED** [review]
Some ideas are nice, but overall concept is not clear for now.

File: concept.py (hunk 3 above)
Comment (ghost) [reply concept.py:13]: 2&3 may be wrapped.
I suppose that `plot()` should exist 'cause it is a native way;
`pyplot.savefig()` doesn't sweep the result away as `pyplot.show()` does.

File: concept.py (hunk 2 above)
Comment (ghost) [reply concept.py:26]: Yep, in this way.

**Comment by ghost** [comment]
I wonder whether you are to combine the approaches or not.

File: concept.py (hunk 3 above)
Comment (gkorepanov) [reply concept.py:13]: The most common use case in my opinion is:
```
    add()
    add()
    add()
    display()
    save()
```

Comment (gkorepanov) [reply concept.py:13]: We can save a plot before showing it, so it is possible

Comment (gkorepanov) [reply concept.py:13]: And I can't imagine a case when we need
`plot(display=False)`

**Comment by gkorepanov** [comment]
Could you please reformulate your question, as I haven't got it?

File: concept.py (hunk 3 above)
Comment (ghost) [reply concept.py:13]: The idea is to not create output file if it is not asked for.

Comment (ghost) [reply concept.py:13]: The most common use expected is to show the plot *and* save it somewhere if the result is OK, and therefore I've proposed to combine these actions.

**Comment by ghost** [comment]
I mean, shall I proceed with such arch or you see it differently?

File: concept.py (hunk 3 above)
Comment (gkorepanov) [reply concept.py:13]: but it's much more letters. Compare:
```
display()
save()
```
and 
```
plot()
plot(output = "smth.png")
```

The former looks cleaner and is more explicit

**Comment by gkorepanov** [comment]
I believe your approach is just fine.

File: concept.py (hunk 3 above)
Comment (ghost) [reply concept.py:13]: Oh, no. The concept implies one has to type
```python
smp.plot('smth.png')
```
and if evrth is OK evrth is done. If wanna only to see the plot, one uses
```python
smp.plot()
```
The exotic case (file output only) is covered as well.

**Comment by ghost** [comment]
/Irrelevant/

**Pull request closed without merging**